### PR TITLE
chore: simplify error name for relay errors

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [MOPLAT-294]

### Description

Relay errors in eigen were reporting with their title as the formatted GraphQL errors from relay. The problem with this is this includes url params and error message, so for almost every request it is distinct leading us to have 1000s of 1 time errors reported in Eigen. This changes the title to a shorter version with just the GraphQL request name and the error if we can parse it out. The full formatted error is still in the metadata of the error.

Old Style:
https://sentry.io/organizations/artsynet/issues/3388128437/?project=5867225&query=is%3Aunresolved&statsPeriod=14d

New Style:
https://sentry.io/organizations/artsynet/issues/3385708903/?project=5867225&query=is%3Aunresolved+is%3Afor_review+assigned_or_suggested%3A%5Bme%2C+none%5D&sort=inbox&statsPeriod=14d

Follow-up:
- After this is merged I will go through sentry and resolve or ignore any errors of the old style, if they are still occurring they should show up in the next released build with the new style error.

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Clean up sentry network error reporting - Brian

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-294]: https://artsyproduct.atlassian.net/browse/MOPLAT-294?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by MounirDhahri** [comment]
Great to see this!

File: src/app/relay/middlewares/errorMiddleware.tests.ts [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+  init: jest.requireActual("@sentry/react-native").init,
+  withScope: jest.requireActual("@sentry/react-native").withScope,
+  captureException: jest.fn(),
+}))
+
+// jest.mock("app/store/GlobalStore", () => ({
+//   GlobalStoreProvider: jest.requireActual("app/store/GlobalStore").GlobalStoreProvider,
+//   GlobalStore: jest.requireActual("app/store/GlobalStore").GlobalStore,
+//   useSelectedTab: jest.fn(() => "home"),
+// }))
```
Comment (brainbicycle) [inline src/app/relay/middlewares/errorMiddleware.tests.ts:16]:
```suggestion
```

File: src/app/relay/createEnvironment.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ import { uploadMiddleware } from "./middlewares/uploadMiddleware"
 export function createEnvironment(
   networkConfig: ConstructorParameters<typeof RelayNetworkLayer> = [
     [
-      // The top middlewares run first, i.e. they are the furtherst from the fetch
+      // The top middlewares run first, i.e. they are the furthest from the fetch
```
Comment (pvinis) [inline src/app/relay/createEnvironment.ts:25]: one the the nicerst fixes I've seen 👌

**Review by pvinis — APPROVED** [review]
very cool! nice to have tests too on this.

**Comment by brainbicycle** [comment]
little follow-up, resolving didn't quite cut it, the old errors continued to show up, needed to add inbound filtering rules to sentry to prevent these errors from showing up again.

**Pull request merged**